#! /bin/sh -e

# DP: Enable gij interpreter on arm

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- src/libjava/configure.host.old	2005-12-31 16:06:00.900551776 +0000
+++ src/libjava/configure.host	2005-12-31 16:06:25.879754360 +0000
@@ -98,6 +98,9 @@
 	enable_hash_synchronization_default=yes
 	slow_pthread_self=yes
 	;;
+  arm*-*)
+	libgcj_interpreter=yes
+	;;
   x86_64-*)
 	sysdeps_dir=x86-64
 	libgcj_flags="${libgcj_flags} -ffloat-store -fno-omit-frame-pointer"
